crates/matrix-sdk/src/test_utils/mocks/mod.rs

```diff
@@ -3230,8 +3230,20 @@ impl<'a> MockEndpoint<'a, LoginEndpoint> {
     pub fn ok(self) -> MatrixMock<'a> {
         self.respond_with(ResponseTemplate::new(200).set_body_json(&*test_json::LOGIN))
     }
+
+    /// Returns a successful response with a given message.
+    pub fn ok_with(self, response: ResponseTemplate) -> MatrixMock<'a> {
+        self.respond_with(response)
+    }
```

Comment:
It would be nice to add an example to this docstring, so newcomers can more easily discover how this should be used. Plenty of methods in the mocks module have examples, feel free to copy most of it from another example and modify it for this method.